src/Http/Middleware/DetermineAndSetStockId.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Rapidez\Msi\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Facades\DB;
+
+class DetermineAndSetStockId
+{
+    /**
+     * @param $request
+     * @param Closure $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
```

Comment:
Perhaps we could cache the result in the user's session or something similar somehow.
I don't expect this to change much while someone is using the website and it saves on a query running every request

Reply:
I updated this part of the PR with a cache layer.

Reply:
@indykoning I added the website_id to the cache key. You raised a good point.